[WMS][12.0] Add stock_location_bin_name (split from stock_location_zone)

grindtildeath opened the pull request:
Followup to #653

grindtildeath committed:
12.0 Add stock_location_bin_name (split from stock_location_zone)

grindtildeath commented on stock_location_bin_name/views/stock_location.xml:
```diff
@@ -0,0 +1,13 @@
+        <field name="inherit_id" ref="stock.view_location_form"/>
+        <field name="arch" type="xml">
+            <field name="usage" position="after">
+                <field name="location_name_format" attrs="{'invisible': [('kind', '=', 'bin')]}" />
```
```suggestion
                <field name="location_name_format" attrs="{'invisible': [('location_kind', '=', 'bin')]}" />
```

grindtildeath committed:
Update stock_location_bin_name/views/stock_location.xml

grindtildeath committed:
fixup! 12.0 Add stock_location_bin_name (split from stock_location_zone)

grindtildeath committed:
[DROPME] oca_dependencies.txt

Commit:
Do not store area field

We only need it for the generation -> useless to
store it. It slows creation of locations for no value.
If we need to search on it, we can use a search method.

grindtildeath committed:
Merge pull request #3 from guewen/12.0_add_stock_location_bin_name

Do not store area field

daramousk commented on stock_location_bin_name/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,2 @@
+This module allows to compute automatically Bin location names based on
```
Generate a readme using https://github.com/OCA/maintainer-tools/blob/master/tools/gen_addon_readme.py for all these.

guewen replied:
not necessary, it will be done by the bot when merged

daramousk commented on stock_location_bin_name/models/stock_location.py:
```diff
@@ -0,0 +1,51 @@
+             "'{area}-{corridor:0>2}.{rack:0>3}"
+             ".{level:0>2}'")
+
+    area = fields.Char(
```
This could also have a `store=True`

guewen replied:
https://github.com/OCA/stock-logistics-warehouse/pull/713/commits/3fcf9b969eb3fa43aa4a6469d5364db0166b865d

Trivedi-Vacha-SerpentCS commented on stock_location_bin_name/models/stock_location.py:
```diff
@@ -0,0 +1,51 @@
+# Copyright 2016-2019 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models, fields, api
```
import statement should be in alphabetical order

Trivedi-Vacha-SerpentCS commented on stock_location_bin_name/models/stock_location.py:
```diff
@@ -0,0 +1,51 @@
+            if location.location_kind == 'area':
+                location.area = location.name
+            else:
+                location.area = location.location_id.area
```
IMHO we can eliminate the else part by defining line 30 before if condition.

guewen replied:
It would access `location.location_id` (so read another record) even when not required.

Commit:
Change the area field as a related field

Follows
https://github.com/OCA/stock-logistics-warehouse/pull/711/commits/5853edb512c66dad75c82bea0c1b17e979a07a7a

guewen commented:
As I added an `area_location_id` in "stock_location_zone", the `area` field can be a related field (https://github.com/OCA/stock-logistics-warehouse/pull/711/commits/5853edb512c66dad75c82bea0c1b17e979a07a7a)

Commit:
Sort imports alphabetically as asked on review

guewen commented:
Continued in 13.0 at #806
